#ifndef DIALOG_H
#define DIALOG_H

#include <QDialog>
#include <QSerialPort>
#include <QSerialPortInfo>
#include <windows.h>

namespace Ui {
class Dialog;
}

class Dialog : public QDialog
{
    Q_OBJECT
public:
    explicit Dialog(QWidget *parent = 0);
    ~Dialog();
    const char enter[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x36,0x34,0x3B,0x33,0x38,0x33,0x30,0x30,0x0D};
    const char reset[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x34,0x36,0x39,0x3B,0x31,0x3B,0x31,0x3B,0x32,0x3B,0x35,0x34,0x38,0x30,0x32,0x0D};
    const char exec1[22] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x34,0x36,0x39,0x3B,0x31,0x3B,0x31,0x3B,0x31,0x3B,0x39,0x37,0x34,0x36,0x0D};
    const char down[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x36,0x36,0x3B,0x36,0x32,0x38,0x37,0x37,0x0D};
    const char up[23] = {0xff,0x3a,0x31,0x3b,0x31,0x31,0x3b,0x33,0x37,0x3b,0x30,0x3b,0x31,0x3b,0x36,0x37,0x3b,0x32,0x36,0x30,0x31,0x32,0x0D};
    const char but0[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x35,0x31,0x3B,0x35,0x30,0x35,0x34,0x33,0x0D};
    const char but1[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x35,0x32,0x3B,0x31,0x33,0x36,0x37,0x39,0x0D};
    const char but2[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x36,0x38,0x3B,0x33,0x38,0x32,0x39,0x37,0x0D};
    const char but3[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x38,0x34,0x3B,0x32,0x32,0x32,0x36,0x39,0x0D};
    const char but4[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x35,0x33,0x3B,0x34,0x32,0x33,0x35,0x30,0x0D};
    const char but5[22] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x36,0x39,0x3B,0x31,0x34,0x33,0x32,0x0D};
    const char but6[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x38,0x35,0x3B,0x35,0x30,0x39,0x34,0x30,0x0D};
    const char but7[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x35,0x34,0x3B,0x33,0x38,0x32,0x35,0x32,0x0D};
    const char but8[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x37,0x30,0x3B,0x33,0x38,0x33,0x35,0x31,0x0D};
    const char but9[23] = {0xFF,0x3A,0x31,0x3B,0x31,0x31,0x3B,0x33,0x37,0x3B,0x30,0x3B,0x31,0x3B,0x38,0x36,0x3B,0x31,0x34,0x30,0x37,0x36,0x0D};
    const char minus[23]= {0xff,0x3a,0x31,0x3b,0x31,0x31,0x3b,0x33,0x37,0x3b,0x30,0x3b,0x31,0x3b,0x35,0x30,0x3b,0x32,0x31,0x38,0x37,0x30,0x0D};
private slots:
    void on_but_enter_clicked();

    void on_but_reset_clicked();

    void on_error(QSerialPort::SerialPortError error);

    void on_but_up_clicked();

    void on_but_down_clicked();

    void on_but_ce_clicked();

    void on_but_0_clicked();

    void on_but_1_clicked();

    void on_but_2_clicked();

    void on_but_3_clicked();

    void on_but_4_clicked();

    void on_but_5_clicked();

    void on_but_6_clicked();

    void on_but_7_clicked();

    void on_but_8_clicked();

    void on_but_9_clicked();

    void on_but_minus_clicked();

private:
    Ui::Dialog *ui;
    QSerialPort * serial;
    bool serial_open;
};

#endif // DIALOG_H
